File: src/main/scala/com/github/zhongl/mockclients/Request.scala
package com.github.zhongl.mockclients

import java.nio.ByteBuffer

/**
 * @author <a href="mailto:zhong.lunfu@gmail.com">zhongl</a>
 */
class Request(dataLength: Int, handleMilliseconds: Int, val responseLength: Int) {

  private val repr = dataLength + "-" + handleMilliseconds + "-" + responseLength

  private val buffer = {
    val direct = ByteBuffer.allocateDirect(dataLength + 12)
    direct.putInt(dataLength + 8)
    direct.putInt(handleMilliseconds)
    direct.putInt(responseLength)
    Utils.times(dataLength) {direct.put('a'.asInstanceOf[Byte])}
    direct.flip
    direct
  }

  def byteBuffer = buffer.asReadOnlyBuffer

  override def toString: String = repr

}

File: src/main/java/com/github/zhongl/nij/netty/perf/Request.java
package com.github.zhongl.nij.netty.perf;

/** @author <a href="mailto:zhong.lunfu@gmail.com">zhongl</a> */
public class Request {
  public final int handleMilliseconds;
  public final int responseLength;

  public Request(int handleMilliseconds, int responseLength) {
    this.handleMilliseconds = handleMilliseconds;
    this.responseLength = responseLength;
  }
}
